src/benchmarks/micro/corefx/System.Threading.Tasks.Dataflow/Perf.Dataflow.cs

```diff
@@ -0,0 +1,219 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Threading.Tasks.Dataflow.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX)]
+    public abstract class PerfTests<T> where T : IDataflowBlock
+    {
+        protected const int MessagesCount = 100_000;
+        protected T block;
+
+        public abstract T CreateBlock();
+
+        [GlobalSetup]
+        public void BlockSetup()
+        {
+            block = CreateBlock();
+        }
+
+        [Benchmark]
+        public async Task Completion()
+        {
+            block.Complete();
+            await block.Completion;
```

Comment:
I'm struggling to understand the setup here.  BlockSetup creates the block, and then this Completion benchmark is going to be invoked a bunch of times, but all times other than the first, it'll already be complete. No?  What is this measuring then?

Reply:
Ditto.